test/integration/buildclient_test.go

```diff
@@ -233,5 +233,5 @@ func (t *testBuildOpenshift) Close() {
 	defer t.lock.Unlock()
 	close(t.stop)
 	t.server.CloseClientConnections()
-	t.server.Close()
+	defer t.server.Close()
```

Comment:
I don't know *why* this would fix anything.  defer isn't magic - it just calls the functions in LIFO order.  Your change is effectively a no-op unless there's something subtly wrong with the go runtime.

The data race on http is still broken upstream... so there's no way to "fix" it here.

----- Original Message -----
> > @@ -233,5 +233,5 @@ func (t *testBuildOpenshift) Close() {
> >  	defer t.lock.Unlock()
> >  	close(t.stop)
> >  	t.server.CloseClientConnections()
> > -	t.server.Close()
> > +	defer t.server.Close()
> 
> Although this might be related to that #731 issue, I'll check it once again.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/989/files#r24586142